crates/holochain_state/src/source_chain.rs

```diff
@@ -1652,47 +1651,42 @@ pub mod tests {
             let action = chain
                 .put_weightless(action_builder, Some(entry), ChainTopOrdering::default())
                 .await?;
-
             chain.flush(&mock).await.unwrap();
-
             (action, entry_hash)
         };
 
-        {
-            let chain = SourceChain::new(
-                db.clone(),
-                dht_db.to_db(),
-                dht_db_cache.clone(),
-                keystore.clone(),
-                alice.clone(),
-            )
-            .await?;
-            // alice should find her own authorship with higher priority than the committed grant
-            // even if she passes in the secret
-            assert_eq!(
-                chain
-                    .valid_cap_grant(function.clone(), alice.clone(), secret.clone())
-                    .await?,
-                Some(CapGrant::ChainAuthor(alice.clone())),
-            );
+        // alice should find her own authorship with higher priority than the
+        // committed grant even if she passes in the secret
+        assert_eq!(
+            chain
+                .valid_cap_grant(function.clone(), alice.clone(), secret.clone())
+                .await?,
+            Some(CapGrant::ChainAuthor(alice.clone())),
+        );
 
-            // bob should be granted with the committed grant as it matches the secret he passes to
-            // alice at runtime
-            assert_eq!(
-                chain
-                    .valid_cap_grant(function.clone(), bob.clone(), secret.clone())
-                    .await?,
-                Some(grant.clone().into())
-            );
-        }
+        // bob should be authorized with transferable cap grant when passing in the secret
+        assert_eq!(
+            chain
+                .valid_cap_grant(function.clone(), bob.clone(), secret.clone())
+                .await?,
+            Some(grant.clone().into())
+        );
```

Comment:
This is maybe unnecessary but feels a little more complete

```suggestion
        // bob and carol (and everyone else) should be authorized with transferable cap grant 
        // when passing in the secret
        assert_eq!(
            chain
                .valid_cap_grant(function.clone(), bob.clone(), secret.clone())
                .await?,
            Some(grant.clone().into())
        );
        assert_eq!(
            chain
                .valid_cap_grant(function.clone(), carol.clone(), secret.clone())
                .await?,
            Some(grant.clone().into())
        );
```